core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -229,9 +229,19 @@ default ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTabl
     /**
      * Gets the metadata for all columns that match the specified table prefix.
      */
+    @Deprecated
     default Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
     {
-        return emptyMap();
+        return listTableColumnsStream(session, prefix)
+                .collect(toMap(ColumnsMetadata::getTable, ColumnsMetadata::getRequiredColumns));
+    }
+
+    /**
+     * Gets the metadata for all columns that match the specified table prefix - as a stream.
+     */
+    default Stream<ColumnsMetadata> listTableColumnsStream(ConnectorSession session, SchemaTablePrefix prefix)
```

Comment:
Why do we return `Stream`? `Stream` in SPI does not look like a good idea. Is it a lazy collection?

Reply:
@kokosing We just needed a redirection-aware method for listing table columns without breaking current SPI, we're just combining the return type change along with it. https://github.com/trinodb/trino/pull/5160#issuecomment-702337998

the current implementation in this PR lets redirection go through `Metadata` instead of it being returned from `ConnectorMetadata` as suggested in the above comment.